powersimdata/scenario/create.py

```diff
@@ -314,35 +318,30 @@ def set_time(self, start_date, end_date, interval):
             self.end_date = end_date
             self.interval = interval
 
+    def print_available_profile(self):
+        """Print available profiles for the grid model
+
+        :raises NotImplementedError: always - implemented in child classes.
+        """
+        raise NotImplementedError("Implemented in the child classes")
+
     def get_base_profile(self, kind):
         """Returns available base profiles.
 
         :param str kind: one of *'demand'*, *'hydro'*, *'solar'*, *'wind'*.
         :return: (*list*) -- available version for selected profile kind.
+        :raises NotImplementedError: always - implemented in child classes.
         """
-        return ProfileInput().get_profile_version(self.grid_model, kind)
+        raise NotImplementedError("Implemented in the child classes")
 
     def set_base_profile(self, kind, version):
```

Comment:
I'm a bit confused. Are these functions supposed to be available in all child classes? Does the _Builder class have to include every function from all its children?

Reply:
We have used this pattern for all parent/child classes, define all methods in the parent class, implement the ones common to all child classes and overwrite the ones that are specific to a child class. I don't have a preference and I can remove from the parent class the methods only available in the `FromCSV` child class. What do you think @jon-hagg ?

Reply:
Makes sense to me - the parent class can just have all the methods _required_ from child classes but if I'm trying to be object oriented I would say it _shouldn't_ know about methods specific to child classes. E.g. suppose another library extends the pattern with new subclasses, we certainly wouldn't update our code based on those packages.

Reply:
Done